### PR TITLE
added type annotations to pl_bolts/optimizers

### LINKED ISSUE #434: Add type annotations to the codebase
## 🚀 Feature
As static type checker `mypy` was added in #433, we can add type annotations to the codebase.

### PR DESCRIPTION
**Pull request by johnpeterson123**
## What does this PR do?
Part of #434

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pl_bolts/optimizers/lr_scheduler.py [hunk 1]
```diff
@@ -68,53 +68,68 @@ def __init__(
...
+                1 + math.cos(
+                    math.pi * (self.last_epoch - self.warmup_epochs) /  # type: ignore[attr-defined]
+                    (self.max_epochs - self.warmup_epochs)
+                )
+            ) /
+            (
+                1 + math.cos(
+                    math.pi * (self.last_epoch - self.warmup_epochs - 1) /  # type: ignore[attr-defined]
+                    (self.max_epochs - self.warmup_epochs)
+                )
```
Comment (Borda) [inline pl_bolts/optimizers/lr_scheduler.py:110]: can we rather specify this as protected func
```python
def _compute_lr(...):
   a = 1 + math.cos(
                    math.pi * (self.last_epoch - self.warmup_epochs) /  # type: ignore[attr-defined]
                    (self.max_epochs - self.warmup_epochs)
   b = 1 + math.cos(
                    math.pi * (self.last_epoch - self.warmup_epochs - 1) /  # type: ignore[attr-defined]
                    (self.max_epochs - self.warmup_epochs)
   return a / b

```

File: pl_bolts/optimizers/lr_scheduler.py [hunk 2]
```diff
@@ -68,53 +68,68 @@ def __init__(
...
         return [
-            self.eta_min + 0.5 * (base_lr - self.eta_min) *
-            (1 + math.cos(math.pi * (self.last_epoch - self.warmup_epochs) / (self.max_epochs - self.warmup_epochs)))
-            for base_lr in self.base_lrs
+            self.eta_min + 0.5 * (base_lr - self.eta_min) * (
+                1 + math.cos(
+                    math.pi * (self.last_epoch - self.warmup_epochs) /  # type: ignore[attr-defined]
+                    (self.max_epochs - self.warmup_epochs)
+                )
+            )
```
Comment (Borda) [inline pl_bolts/optimizers/lr_scheduler.py:133]: the same here def as internal func

**Comment by Borda** [comment]
@johnpeterson123 mind check review comments and conflicts? :rabbit:

**Comment by Borda** [comment]
@johnpeterson123 how is it going here? mind resolve conflicts? :rabbit:

**Comment by Borda** [comment]
@SeanNaren mind check it if we can quick let it land, otherwise we may close it as @johnpeterson123 is not responding

**Comment by SeanNaren** [comment]
@Borda not sure we should add so many ignores as @akihironitta mentions in https://github.com/PyTorchLightning/lightning-bolts/issues/434#issuecomment-810287910

Is there a way to do this typing without all these ignores?

**Comment by akihironitta** [comment]
Feel free to reopen this PR if needed.

**Pull request closed without merging**